scala-xml/src/main/scala/org/http4s/scalaxml/ElemInstances.scala

```diff
@@ -67,4 +71,25 @@ trait ElemInstances {
       }
     }
   }
+
+  implicit def xmlDecoder[F[_]](implicit F: Async[F]): EntityDecoder[F, Elem] = {
+    import EntityDecoder._
+    decodeBy(MediaType.text.xml, MediaType.text.html, MediaType.application.xml) { msg =>
+      val source = new InputSource()
+      msg.charset.foreach(cs => source.setEncoding(cs.nioCharset.name))
+
+      collectBinary(msg).flatMap[DecodeFailure, Elem] { chunk =>
+        source.setByteStream(new ByteArrayInputStream(chunk.toArray))
```

Comment:
We could avoid this strictness by converting the fs2 stream to an InputStream.  I think we'd spend CPU to save memory with that approach.